src/schemapack/_internals/isolate.py

```diff
@@ -92,7 +92,7 @@ def identify_resource_dependencies(  # noqa: C901, PLR0912
     resource_blacklist: dict[ClassName, set[ResourceId]] = defaultdict(set)
     resource_blacklist[class_name].add(resource_id)
     if _resource_blacklist:
-        for class_name, resource_ids in _resource_blacklist.items():
+        for class_name, resource_ids in _resource_blacklist.items():  # noqa: PLR1704
```

Comment:
`class_name` should not shadow the variable that was passed into the function.
This can lead to subtle issues, i.e. in line 141 `dependencies_by_class[class_name].add(resource_id)` likely references the wrong class_name, if `_resource_blacklist` is not empty

Reply:
Seems like an inherited issue. Let's look it over separately.

Reply:
Issue opened here: https://github.com/ghga-de/schemapack/issues/37